external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -4130,40 +4130,64 @@ fn parse_use_decl(
             }
             let address_start_loc = context.tokens.start_loc();
             let address = parse_leading_name_access(context)?;
-            consume_token_(
+            let colon_colon_loc = context.tokens.current_token_loc();
+            if let Err(diag) = consume_token_(
                 context.tokens,
                 Tok::ColonColon,
                 start_loc,
                 " after an address in a use declaration",
-            )?;
-            match context.tokens.peek() {
-                Tok::LBrace => {
-                    let parse_inner = |ctxt: &mut Context<'_, '_, '_>| {
-                        let (name, _, use_) = parse_use_module(ctxt)?;
-                        Ok((name, use_))
-                    };
-                    let use_decls = parse_comma_list(
-                        context,
-                        Tok::LBrace,
-                        Tok::RBrace,
-                        &TokenSet::from([Tok::Identifier]),
-                        parse_inner,
-                        "a module use clause",
-                    );
-
-                    Use::NestedModuleUses(address, use_decls)
-                }
-                _ => {
-                    let (name, end_loc, use_) = parse_use_module(context)?;
-                    let loc = make_loc(context.tokens.file_hash(), address_start_loc, end_loc);
-                    let module_ident = sp(
-                        loc,
-                        ModuleIdent_ {
-                            address,
-                            module: name,
-                        },
-                    );
-                    Use::ModuleUse(module_ident, use_)
+            ) {
+                context.env.add_diag(*diag);
+                Use::Partial(address, None, None)
+            } else {
+                match context.tokens.peek() {
+                    Tok::LBrace => {
+                        let lbrace_loc = context.tokens.current_token_loc();
+                        // if next token after `{` is not `}` then it means that there are module
+                        // uses that could potentially be parsed
+                        let module_uses_to_parse_exist = !context
+                            .tokens
+                            .lookahead()
+                            .map(|tok| tok == Tok::RBrace)
+                            .unwrap_or_default();
```

Comment:
Can you explain why this is necessary?

Reply:
The idea here is that an empty list of module uses is valid code but it probably should't be marked as "incomplete" as this implies a parsing error (and may affect how IDE handles this case). In other words, trying to provide as accurate info here as possible thinking that it may be useful to the consumer of this info

Reply:
As it happens, it is actually better for the IDE if empty lists are treated as `Partial`s, and it does not matter for the compiler as the empty list contains no useful information anyway so can be safely ignored as part of `Partial`. It's a bit of a long story why this helps the IDE but the short version is that at parsing AST we no longer have information about locations of `::` tokens and have to jump through some hoops to make sure that we are auto-completing for the right `::`.